x-pack/test/apm_api_integration/common/utils/create_and_run_apm_ml_jobs.ts

```diff
@@ -5,11 +5,35 @@
  * 2.0.
  */
 
+import { Client } from '@elastic/elasticsearch';
 import job from '@kbn/ml-plugin/server/models/data_recognizer/modules/apm_transaction/ml/apm_tx_metrics.json';
 import datafeed from '@kbn/ml-plugin/server/models/data_recognizer/modules/apm_transaction/ml/datafeed_apm_tx_metrics.json';
 import { MlApi } from '../../../functional/services/ml/api';
 
-export function createAndRunApmMlJob({ ml, environment }: { ml: MlApi; environment: string }) {
+export async function createAndRunApmMlJobs({
+  es,
+  ml,
+  environments,
+}: {
+  es: Client;
+  ml: MlApi;
+  environments: string[];
+}) {
+  if (environments.length === 0) {
+    return;
+  }
+
+  // Creating multiple ml jobs in parallel is causing this tests to be flaky
+  // https://github.com/elastic/elasticsearch/issues/36271
+  await createAndRunApmMlJob({ environment: environments.shift() as string, ml });
```

Comment:
can we copy the array first before mutating it?

Reply:
Got rid of the shifting. In favour of simplicity we are checking the status always so we can avoid that logic

Reply:
but we are now creating them in parallel again?